frontend/src/hooks/refresh.ts

```diff
@@ -0,0 +1,61 @@
+import { useSelector } from "react-redux";
+import { KialiAppState } from "../store/Store";
+import { IntervalInMilliseconds, TimeInMilliseconds } from "../types/Common";
+import { useEffect, useState } from "react";
+
+let numSubscribers = 0;
+let intervalId: null | number = null;
+
+function doTick(time?: TimeInMilliseconds) {
+  const refreshTick = new CustomEvent('refreshTick', {detail: time ?? Date.now()});
+  document.dispatchEvent(refreshTick);
+}
+
+export function triggerRefresh(time?: TimeInMilliseconds) {
+  doTick(time);
+}
```

Comment:
Do we need `triggerRefresh` if we also have the `connectRefresh` HOC? Could we pass `setLastRefreshAt` through there?

Reply:
Well.... setLastRefreshAt is no longer in Redux. So, in that `connectRefresh` we can pass a `setLastRefreshAt` prop, but it would be only a reference to `triggerRefresh`.

I'm hoping some day we would remove that `connectRefresh` HOC. So, I think we still would need this.